### PR TITLE
Improve naming in term canonization when handling HO variables

### PR DESCRIPTION
**Pull request by HanielB**
Previously all higher-order variables would me named with "-" followed by the index, since the method did not account for functional types (which are printed as with `(-> ...)`). This commit changes it so that it takes the return type, which will always be atomic, thus giving more meaningful names to the canonical variables.

### CODE REVIEW

File: src/expr/term_canonize.cpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ Node TermCanonize::getCanonicalFreeVar(TypeNode tn, unsigned i, uint32_t tc)
   while (tvars.size() <= i)
   {
     std::stringstream oss;
-    oss << tn;
```
Comment (ajreynol) [inline src/expr/term_canonize.cpp:108]: Can you perhaps add "fun_" or something to the name here?

Comment (ajreynol) [reply src/expr/term_canonize.cpp:108]: Better, can you make the name simply `f`?

Comment (HanielB) [reply src/expr/term_canonize.cpp:108]: You mean for all functional-type vars to have the names as `f` ++ `i`?

Comment (ajreynol) [reply src/expr/term_canonize.cpp:108]: Yes

Comment (HanielB) [reply src/expr/term_canonize.cpp:108]: OK, done

**Review by ajreynol — APPROVED** [review]

**Pull request merged**